clang/lib/Driver/Driver.cpp

```diff
@@ -2187,7 +2187,16 @@ void Driver::BuildInputs(const ToolChain &TC, DerivedArgList &Args,
           if (!Args.hasArgNoClaim(options::OPT_E) && !CCCIsCPP())
             Diag(IsCLMode() ? clang::diag::err_drv_unknown_stdin_type_clang_cl
                             : clang::diag::err_drv_unknown_stdin_type);
-          Ty = types::TY_C;
+          if (memcmp(Value, "-", 2) == 0) { 
```

Comment:
I don't get it. Considering the `if` condition above (line 2181), is this condition ever false?

Reply:
@pawosm-arm This condition is false when a real file is given to the input (not stdin).
Btw- this condition was done before- i did not make it.

Reply:
the code is fixed. The `if (memcmp(Value, "-", 2) == 0)` condition was doubled.
Now one of them is removed.Please take a look.Thanks in advance